Migration guide — Wazuh Dashboard 4.x to 5.x (#8553)
wazuh/wazuh-kibana-app: docs/guide/migration/README.md

Reviewer: **suggestion**: the `wazuh.yml` and `reports` files are under the same parent directory, you could provide a command to copy all the `wazuh` (data/wazuh) directory.

This approach reduces the steps.

Author: Solved https://github.com/wazuh/wazuh-dashboard-plugins/commit/0d9de86941ff83d68e2b96ef0e79fb737f892380

Reviewer: **thought**:  the file contains information that is commented in the specific docs, we could consider avoid the instructions duplication.

| Plugin structure                                | `wazuh` + `wazuh-core` + `wazuh-check-updates` (split introduced in 4.x)          | `wazuh` (main) + `wazuh-core` + `wazuh-check-updates`                                                                                                                          |
| Plugin configuration file                       | `wazuh.yml`                                                                       | `opensearch_dashboards.yml`                                                                                                                                                    |
| Wazuh server host setting key                   | `hosts[].id`                                                                      | `wazuh_core.hosts.<name>`                                                                                                                                                      |

> | Default index pattern                           | `wazuh-alerts-*`                                                                  | `wazuh-events*`                                                                                                                                                                |

Reviewer 2: default index pattern is wazuh-events-v5*
```suggestion
| Default index pattern                           | `wazuh-alerts-*`                                                                  | `wazuh-events-v5*`                                                                                                                                                                |
```

Author: Resolved https://github.com/wazuh/wazuh-dashboard-plugins/pull/8553/changes/40e51d85dd2898d81787ce10c700b2a3187e0d66

| Health check individual toggles (`checks.*`)    | Individual boolean settings per check                                             | Single `healthcheck.checks_enabled` regex/list                                                                                                                                 |

> | Statistics indices (`wazuh-statistics-*`)       | Collected automatically via `cron.*` tasks                                        | Removed; data model replaced by `wazuh-states-*` indices                                                                                                                       |

Reviewer 2: ```suggestion
| Statistics indices (`wazuh-statistics-*`)       | Collected automatically via `cron.*` tasks                                        | Removed; data model replaced by `wazuh-metrics-comms` and `wazuh-metrics-normalization` indices                                                                                                                       | 
```

Author: Resolved https://github.com/wazuh/wazuh-dashboard-plugins/pull/8553/changes/40e51d85dd2898d81787ce10c700b2a3187e0d66

> | Agent monitoring indices (`wazuh-monitoring-*`) | Collected automatically via `wazuh.monitoring.*`                                  | Removed; agent status queried on demand from API                                                                                                                               |

Reviewer 2: `wazuh-metrics-agents` is also now available

1. [Configuration migration](./configuration.md) — Translate `wazuh.yml` settings to `opensearch_dashboards.yml`.
2. [Custom dashboards and visualizations](./dashboards.md) — Export and re-import saved objects.
3. [Reports](./reports.md) — Preserve existing PDF reports generated by the 4.x plugin.
4. [Multi-manager environments](./multi-manager.md) — Adapt environments previously configured with multiple Wazuh manager API connections.

## Before you begin

Complete the following steps before performing any migration task:

### Back up the configuration file

The 4.x plugin configuration file is located under the `path.data` directory defined in `opensearch_dashboards.yml` (default: `/usr/share/wazuh-dashboard/data`).

Run the following command **on the 4.x dashboard server**:

```bash
sudo cp /usr/share/wazuh-dashboard/data/wazuh/config/wazuh.yml \
   ~/wazuh-yml-backup-$(date +%Y%m%d).yml
```

### Back up the dashboard configuration

Run the following command **on the 4.x dashboard server**:

```bash
sudo cp /etc/wazuh-dashboard/opensearch_dashboards.yml \
   ~/opensearch-dashboards-yml-backup-$(date +%Y%m%d).yml

> ```

Reviewer: **suggestion**: unify the backup of files under a backup directory (include date).

Author: Solved https://github.com/wazuh/wazuh-dashboard-plugins/pull/8553/changes/0d9de86941ff83d68e2b96ef0e79fb737f892380

### Export custom saved objects

Export only the dashboards and visualizations you created or modified. Default Wazuh objects are re-provisioned automatically in 5.x and must not be re-imported.

1. In the Wazuh dashboard, navigate to **☰ Menu > Dashboard Management > Saved objects**.
2. Select the checkboxes next to each custom dashboard or visualization and click **Export**.
3. Enable **Include related objects** and save the resulting `.ndjson` file to a secure location.

If you export all objects as a fallback, use the **Check for existing objects** conflict strategy when importing into 5.x. See [Custom dashboards and visualizations](./dashboards.md) for details.

> Alternatively, use the API. Run the following command from **any machine with network access to the 4.x dashboard**, replacing `<DASHBOARD_HOST>` with the 4.x dashboard hostname or IP and `<PASSWORD>` with the admin password. The output file is saved in the current working directory:

Reviewer: ```suggestion
Alternatively, use the API. Run the following command from **any machine with network access to the 4.x dashboard**, replacing `<DASHBOARD_HOST>` with the 4.x dashboard hostname or IP, `<DASHBOARD_PORT>` with the port and `<PASSWORD>` with the admin password. The output file is saved in the current working directory:
```

Author: Solved https://github.com/wazuh/wazuh-dashboard-plugins/commit/0d9de86941ff83d68e2b96ef0e79fb737f892380

```bash

> curl -X POST "https://<DASHBOARD_HOST>:5601/api/saved_objects/_export" \

Reviewer: ```suggestion
curl -X POST "https://<DASHBOARD_HOST>:<DASHBOARD_PORT>/api/saved_objects/_export" \
```

Author: Solved https://github.com/wazuh/wazuh-dashboard-plugins/commit/0d9de86941ff83d68e2b96ef0e79fb737f892380

  -H "osd-xsrf: true" \
  -H "Content-Type: application/json" \
  -u admin:<PASSWORD> \

>   --cacert /etc/wazuh-dashboard/certs/root-ca.pem \

Reviewer: **suggestion**: remove the usage of certs in the command, this is using a certificate that is only in the dashboard and the previous paragraph mentions to
> Run the following command from **any machine with network access to the 4.x dashboard**

If the command is executed from another machine, the file could not exist.

```suggestion
  -k \
```

Author: Solved https://github.com/wazuh/wazuh-dashboard-plugins/commit/0d9de86941ff83d68e2b96ef0e79fb737f892380

  -d '{"type": ["dashboard", "visualization", "search", "index-pattern"], "includeReferencesDeep": true}' \
  -o saved-objects-backup-$(date +%Y%m%d).ndjson
```

### Download existing reports

PDF reports generated by the 4.x plugin are stored per user on the dashboard server filesystem. Each user's reports are in a subdirectory named after a hash of their username:

```
<path.data>/wazuh/downloads/reports/<hashed_username>/
```

On a standard installation, the base directory is:

```
/usr/share/wazuh-dashboard/data/wazuh/downloads/reports/
```

Each user's PDFs are stored in a subdirectory named after a hash of their username (for example, `/usr/share/wazuh-dashboard/data/wazuh/downloads/reports/<hashed_username>/`).

Copy the entire `reports/` directory (including all per-user subdirectories) to a safe location before uninstalling the 4.x package. Run the following command **on the 4.x dashboard server**:

```bash
sudo cp -r /usr/share/wazuh-dashboard/data/wazuh/downloads/reports/ \

>    ~/wazuh-reports-backup-$(date +%Y%m%d)/

Reviewer: If unified the backup directory, adapt the destination in the command.

Author: Solved https://github.com/wazuh/wazuh-dashboard-plugins/commit/0d9de86941ff83d68e2b96ef0e79fb737f892380
